apps/nextjs-website/src/lib/api.ts

```diff
@@ -39,3 +46,68 @@ export const getHeaderProps = async (): Promise<HeaderProps> => {
   const navigation = await getNavigation('main-navigation', appEnv);
   return makeHeaderProps(navigation, header);
 };
+
+const generateLinks = (
+  section: FooterSection,
+  isSocialLink = false
+): ReadonlyArray<FooterLinksType> => {
+  const filteredLinks = section.links.filter((link) =>
+    isSocialLink
+      ? link.linkType === 'social' && link.icon
+      : ['internal', 'external'].includes(link.linkType)
+  );
+
+  return filteredLinks.map((link) => ({
+    label: link.text ?? link.href,
+    href: link.href,
+    ariaLabel: link.ariaLabel ?? link.href,
+    linkType: link.linkType as LinkType,
+    ...(isSocialLink && { icon: link.icon as keyof typeof MuiIcons }),
```

Comment:
Is the cast required? It compiles without it

```suggestion
    ...(isSocialLink && { icon: link.icon }),
```